app/controllers/tags_controller.rb

```diff
@@ -90,23 +88,34 @@ def show
   end
 
   def feed
-    begin
-      @tag = Tag.find(params[:id])
-    rescue ActiveRecord::RecordNotFound
-       raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
-    end
-    if !@tag.canonical? && @tag.merger
-      @tag = @tag.merger
-    end
-    # Temp for testing
-    if %w(Fandom Character Relationship).include?(@tag.type.to_s) || @tag.name == "F/F"
-      if @tag.canonical?
-        @works = @tag.filtered_works.visible_to_all.order("created_at DESC").limit(25)
+    hash = Rails.cache.fetch(Tag.tag_feeds_key(params[:id])) do
+      path = nil
+      works = nil 
+      begin
+        tag = Tag.find(params[:id])
+      rescue ActiveRecord::RecordNotFound
+        raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
+      end
+      if !tag.canonical? && tag.merger
+        tag = tag.merger
+      end
+      # Temp for testing
+      if %w(Fandom Character Relationship).include?(tag.type.to_s) || tag.name == "F/F"
+        if tag.canonical?
+          works = tag.filtered_works.visible_to_all.order("created_at DESC").limit(25).all
+        else
+          works = tag.works.visible_to_all.order("created_at DESC").limit(25).all
+        end
       else
-        @works = @tag.works.visible_to_all.order("created_at DESC").limit(25)
+        path = tag_works_path(tag_id: tag.to_param)
       end
-    else
-      redirect_to tag_works_path(:tag_id => @tag.to_param) and return
+      { path: path, works: works, tag: tag }
+    end
+    redirect = hash[:path]
+    @works = hash[:works]
+    @tag = hash[:tag]
+    unless redirect.nil? 
+      redirect_to path and return
     end
```

Comment:
What I mean is, the line "redirect_to path and return" needs to be exercised by a test and is not, because I am pretty sure it SHOULD be failing given the way you have it:

```
hash = Rails.cache.fetch(Tag.tag_feeds_key(params[:id])) do
  ....
  { path: path, works: works, tag: tag }
end
```

Here you are setting a variable "hash". But the variable "path" itself is ONLY set inside the cached block--which will not run if the value is retrieved from the cache. You do set `redirect = hash[:path]` but then you don't actually USE the variable "redirect", you use the variable "path" which has not been set. 

So when you do `redirect_to path and return` where the value is coming out of the hash, you should be getting a variable not set failure. Since you are NOT getting that failure out of the tests, that tells me the tests aren't exercising this line, and you want a test that does exercise that line. That line gets invoked when redirect (which is equal to hash[:path]) is not nil AND when the value comes out of the cache, so you need to create a test case where you are redirecting via a cached value.